### ISSUE TITLE
Don't name-mangle buffer names

### ISSUE THREAD

**Issue by BearishSun** [LukasBanana/XShaderCompiler#34]
Currently when a `StructuredBuffer` is converted into a `buffer` block, its name is mangled by appending "xst_" prefix (by default). This mangling makes it hard to write high level render API agnostic code (e.g. a system that transparently supports DirectX, OpenGL and Vulkan). I suggest we instead mangle the buffer contents (the array variable), since that bit is not important for external code.

**Comment by LukasBanana** [comment]
I'll add it as option when I start refactoring the name-mangling shell commands.
I want to reduce the amount of shell commands to something like with the `-W` option with different warning types:
```
-P<TYPE>=VALUE
    Sets the prefix of the specified name-mangling type; valid values:
      in  => Shader input prefix
      out => Shader output prefix
      ...

-N<TYPE> [ON|OFF]
    Enables/disables the specified name-mangling option; default=OFF; valid values:
      buffer-wrapper   => Renames the outer 'buffer' object instead of the inner field.
      always-semantics => Always use semantics for input/output variables.
      ...
```
Then you could enable this feature with `xsc -Nbuffer-wrapper ON`.

**Comment by BearishSun** [comment]
Sounds good!

**Comment by LukasBanana** [comment]
Solved with [e0135b1](https://github.com/LukasBanana/XShaderCompiler/commit/e0135b101a014ba624c9e70e342c12a73a22292c).

I reduced all name-mangling and prefix shell commands to `-P<TYPE>` and `-N<TYPE>` commands, and implemented the name-mangling functionality just like your proposal.

Your request can be achieved with `xsc -Nbuffer-fields ON` for instance.

**Comment by BearishSun** [comment]
Great, thanks!
